engine/table/src/test/java/io/deephaven/engine/table/impl/TestTableUpdateValidator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2016-2019 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.engine.context.QueryScope;
+import io.deephaven.engine.exceptions.TableAlreadyFailedException;
+import io.deephaven.engine.rowset.RowSetShiftData;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.testutil.*;
+import io.deephaven.engine.testutil.generator.DoubleGenerator;
+import io.deephaven.engine.testutil.generator.FloatGenerator;
+import io.deephaven.engine.util.PrintListener;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.junit.Test;
+
+import java.util.Random;
+
+import static io.deephaven.engine.testutil.TstUtils.*;
+import static io.deephaven.engine.util.TableTools.intCol;
+
+/**
+ * Test that the TableUpdateValidator can be put in the middle of an operation.
+ */
+public class TestTableUpdateValidator extends QueryTableTestBase {
+    @Test
```

Comment:
If you feel like it, you could delete these annotations, since IntelliJ doesn't like them on a JUnit3 style test.